cdist/exec/local.py

```diff
@@ -238,7 +239,7 @@ def run_script(self, script, env=None, return_output=False,
                         message_prefix=message_prefix)
 
     def save_cache(self):
-        destination = os.path.join(self.cache_path, self.hostdir)
+        destination = os.path.join(self.cache_path, self.target_host)
```

Comment:
Darko Poljak wrote on 07/07/16 17:52:
>> @@ -238,7 +239,7 @@ def run_script(self, script, env=None, return_output=False,
>>                          message_prefix=message_prefix)
>>
>>      def save_cache(self):
>> -        destination = os.path.join(self.cache_path, self.hostdir)
>> +        destination = os.path.join(self.cache_path, self.target_host)
>
> @asteven Why self.target_host instead of self.hostdir at line 242? Wouldn't this try to create directory path that contain target_host?

In the cache I doen't care. It doesn't hurt anybody there as it's not 
used by any code/shell.

They reason that this all broke for me was that my target_host is 
sometimes a string like 
chroot://some.server.com/path/to/chroot/on/that/server.

The way you implemented it this string 
(chroot://some.server.com/path/to/chroot/on/that/server) ends up in 
CDIST_PATH where the extra colons (:) break the cdist-conf-dirs from 
properly populating which leads to types not being found at runtime.